src/Microsoft.OData.Core/Json/BufferingJsonReader.cs

```diff
@@ -474,6 +681,150 @@ protected virtual void ProcessObjectValue()
+                        result = await this.ReadNextAndCheckForInStreamErrorAsync()
+                            .ConfigureAwait(false);
+                    }
+                }
+            }
+            else
+            {
+                if (this.bufferedNodesHead == null)
+                {
+                    // if parsingInStreamError nothing in the buffer; read from the base,
+                    // else read the next node from the input stream and check
+                    // whether it is an in-stream error
+                    result = this.parsingInStreamError ?
+                        await this.asyncInnerReader.ReadAsync().ConfigureAwait(false) :
+                        await this.ReadNextAndCheckForInStreamErrorAsync().ConfigureAwait(false);
+                }
+                else
+                {
+                    Debug.Assert(!this.parsingInStreamError, "!this.parsingInStreamError");
+
+                    // non-buffering read from the buffer
+                    result = this.bufferedNodesHead.NodeType != JsonNodeType.EndOfInput;
+                    this.removeOnNextRead = true;
+                }
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Called asynchronously whenever we find a new object value in the payload.
+        /// The base class implementation reads ahead and tries to parse it as an in-stream error payload. If it finds one it will throw it.
+        /// </summary>
+        /// <returns>A task that represents the asynchronous read operation.</returns>
+        /// <remarks>
+        /// This method is called when the reader is in the buffering mode and can read ahead (buffering) as much as it needs to
+        /// once it returns the reader will be returned to the position before the method was called.
+        /// The reader is always positioned on a start object when this method is called.
+        /// </remarks>
+        protected virtual async Task ProcessObjectValueAsync()
+        {
+            Debug.Assert(this.currentBufferedNode.NodeType == JsonNodeType.StartObject, "this.currentBufferedNode.NodeType == JsonNodeType.StartObject");
+            Debug.Assert(this.parsingInStreamError, "this.parsingInStreamError");
+            this.AssertBuffering();
+            this.AssertAsynchronous();
+
+            // Only check for in-stream errors if the buffering reader is told to do so
+            if (!this.DisableInStreamErrorDetection)
+            {
+                // Move to the first property of the potential error object (or the end-object node if no properties exist)
+                await this.ReadInternalAsync()
+                    .ConfigureAwait(false);
+
+                // We only consider this to be an in-stream error if the object has a single 'error' property
+                bool errorPropertyFound = false;
+                Tuple<bool, ODataError> readInStreamErrorPropertyResult = null;

```

Comment:
```suggestion
               (bool hasError, ODataError error) readInStreamErrorPropertyResult = null;
```
Naming the fields of the tuple makes it easier to refer to them, and avoids using `Item1`, `Item2`, et

Reply:
@habbes Named tuples are not support in net45 and netstandard1.1. We could use the `#if NETSTANDARD2_0` directives but it'd be too messy. I'd rather we adopt named tuples after we abandon support for the 2 standards I referenced